### ISSUE TITLE
'Collection was modified' message appears in PMC when opening a solution

### ISSUE THREAD

**Issue by emgarten** [NuGet/Home#657]
Steps:
1. Install the 3.0-rtm 429 build of NuGet
2. Open PMC
3. Delete out the NuGet packages from the NuGet.PackageManagement Packages folder
4. Open NuGet.PackageManagement

"Collection was modified; enumeration operation may not execute." appears in PMC

**Comment by KenKilty** [comment]
I have observed repro of this bug with extremely large solutions (200+ .csproj's for example) when a PM Console command is run before Visual Studio has fully loaded all projects.  Running a command that is solution wide such as Update-Package <packageName> shortly after opening such a large project will reproduce the bug.  I suspect that when the cmdlet is run a List<T> or other collection is captured, an iterator begins to process the collection, and additional projects are attempted to be added to the underlying collection while the iterator is running.  While the customer repro I observed with 200+ projects is a bit extreme I am sure it's possible to repro this behavior with smaller projects as you are constrained to the speed of the clients (Visual Studio's) ability to load all projects associated with a solution.
